lisa/tools/meson.py

```diff
@@ -30,24 +34,75 @@ def can_install(self) -> bool:
     def _install(self) -> bool:
         posix_os: Posix = cast(Posix, self.node.os)
         # use pip to make sure we install a recent version
-        if (not posix_os.package_exists("meson")) or posix_os.get_package_information(
-            "meson", use_cached=False
-        ) < "0.52.0":
-            username = self.node.tools[Whoami].get_username()
-            self.node.tools[Pip].install_packages("meson", install_to_user=True)
-            # environment variables won't expand even when using shell=True :\
-            self.node.tools[Ln].create_link(
-                f"/home/{username}/.local/bin/meson", "/usr/bin/meson", force=True
-            )
-            # ensure sudo has access as well
-            self.node.execute(
-                "pip3 install meson",
-                sudo=True,
-                shell=True,
-                no_debug_log=True,
-                no_info_log=True,
-                no_error_log=True,
-            )
+
+        package_installed = ""
+        package_available = ""
+        # packaged as 'meson' on older systems and 'python3-meson' on newer ones,
+        # since it's actually just a python package.
+        # But now we have a bunch of annoying cases.
+        # 1.  meson is installed and it's the right version
+        # 2. 'meson' is installed but the wrong version
+        # 3. meson is not installed and the right version is in the repo
+        # 4. meson is not installed and the wrong version is in the repo
+
+        for pkg in [
+            "python3-meson",
+            "meson",
+        ]:
+            if posix_os.package_exists(pkg):
+                package_installed = pkg
+            if posix_os.is_package_in_repo(pkg):
+                package_available = pkg
+            if package_installed or package_available:
+                break
+
+        if package_installed:
+            # check the installed version before touching anything
+            if (
+                posix_os.get_package_information(pkg, use_cached=False)
+                >= self._minimum_version
+            ):
+                # meson is installed and it's the right version
+                return self._check_exists()
+
+        # otherwise, install the available package from the repo
+        if package_available:
+            posix_os.install_packages(package_available)
+            # and update the cached version info
+            posix_os.get_package_information(pkg, use_cached=False)
+            package_installed = package_available
+
+        # check the version, return if it's good, remove if not
+        if package_installed:
```

Comment:
This logic should be merged with previous `if package_installed`. It's like: 

```
if package_installed:
    if version is ok:
        return check exists
    else:
        uninstall

if package_available:
    install (assume the version is ok.)
```

Reply:
It doesn't assume the version is valid, so my logic uninstalls the package if the version is invalid. **else: uninstall**

Reply:
@squirrelsc updated, now we just install if there's an update then check the version once (and uninstall if it doesn't meet the requirements)

Reply:
It seems to be missing a login step to reinstall the correct version. If v2 is expected but v1 is installed, can it uninstall v1 and install v2?

Reply:
Installing a (later) version from the package manager will implicitly update the version to a later version. You don't need to call apt remove before upgrading a package. If one is available, it will be a later version.
The 'installed but correct version' case is already covered before we get to _install(). The tool will call _check_exists.
If none of that works, we do the less desireable workaround of messing with pip.